torch/csrc/jit/codegen/cuda/ir_base_nodes.cpp

```diff
@@ -193,16 +193,17 @@ bool Val::isConstScalar() const {
 }
 
 bool Val::isConstInt() const {
-  if (!isAnInt()) {
-    return false;
-  }
-  return ConstCheck::isConstInt(this);
+  return isConstExpr() && isAnInt();
+}
+
+bool Val::isConstExpr() const {
+  return ConstCheck::isConst(this);
```

Comment:
Do we need to separate it to the two functions?

The name of `Val::isConstExpr()` doesn't sound quite right as `Val` is not an expression.

Reply:
Agree, removed